packages/azure-openapi-validator/autorest/src/spectral-plugin-func.ts

```diff
@@ -90,189 +65,114 @@ export async function spectralPluginFunc(initiator: IAutoRestPluginInitiator): P
-        // https://github.com/stoplightio/spectral/blob/44c40e2b7c9ea6222054da8700049b0307cc5f8b/packages/core/src/ruleset/rule.ts#L121
-        // Resulting in value of -1, as defined here:
-        // https://github.com/stoplightio/spectral/blob/44c40e2b7c9ea6222054da8700049b0307cc5f8b/packages/ruleset-migrator/src/transformers/rules.ts#L39
-        rule.severity = "off"
-        // We must disable the rule to ensure it is not run at all. Spectral source for that is in [1].
-        // Otherwise, if it throws due to a bug in rule implementation, it will result in fatal error.
-        // Example of how a rule can thrown is given in [2].
-        // Example where we disable a rule from running in production so it doesn't throw is given in [3].
-        // Without this line, the fix in [3] doesn't help. For a proof of that, see [4] and its log, [5],
-        // showing that a rule that is supposed to run in stagingOnly, still throws in production.
-        // [1] https://github.com/stoplightio/spectral/blob/6d0991572f185ce5cf4031dc1d8eb4035b5eaf1d/packages/core/src/runner/runner.ts#L39
-        // [2] https://github.com/Azure/azure-openapi-validator/pull/595
-        // [3] https://github.com/Azure/azure-openapi-validator/pull/596
-        // [4] https://github.com/Azure/azure-rest-api-specs/pull/26024
-        // [5] https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3125612&view=logs&j=0574a2a6-2d0a-5ec6-40e4-4c6e2f70bea2&t=80c3e782-49f0-5d1c-70dd-cbee57bdd0c7&l=252
-        rule.enabled = false
-      }
+      Text: `spectralPluginFunc: Ignoring file matching to 'common-types/resource-management': '${openApiSpecFile}'`,
     })
+    return
   }
-}
 
-function printRuleNames(initiator: IAutoRestPluginInitiator, ruleset: Ruleset, resolvedOpenapiType: OpenApiTypes) {
-  const ruleNames: string[] = Object.keys(ruleset.rules)
-    // Case-insensitive sort.
-    // Source: https://stackoverflow.com/a/60922998/986533
-    .sort(Intl.Collator().compare)
+  try {
+    const openApiSpecFilePath = openApiSpecFile.startsWith("file:///") ? fileURLToPath(openApiSpecFile) : openApiSpecFile
+    const openApiSpecContent: string = await readFileUsingCache(initiator, openApiSpecFile)
+    // safeLoad() documented at: https://github.com/nodeca/js-yaml/tree/v3?tab=readme-ov-file#safeload-string---options-
+    // Empirically confirmed the returned value type is object, not string.
+    const openApiSpecYml: any = safeLoad(openApiSpecContent)
+    // "x-typespec-generated" is expected to be found at JSONPath of $.info.x-typespec-generated.
+    // Example: https://github.com/Azure/azure-rest-api-specs/blob/fca48bec19cc5aab0a45c0769bfca0f667164dbf/specification/edgemarketplace/resource-manager/Microsoft.EdgeMarketplace/stable/2023-08-01/operations.json#L7
+    const specIsGeneratedFromTypeSpec = Boolean(openApiSpecYml.info["x-typespec-generated"]) // JSON.stringify(openApiSpecYml).includes("x-typespec-generated")
```

Comment:
Do we need to handle the case where before was not typespec but the after is typespec now?

Reply:
I'm not sure what we want to do here but perhaps we can check to see if the after is typespec and disable the rules for both the before and after runs.

Reply:
@weshaggard from the point of view of LintDiff I believe we won't ever produce any extra errors. If a spec was converted to TypeSpec it can only have **less** errors reported, because we are only **disabling** rules for TypeSpec case. If I understand @tjprescott correctly, there might be a case the new spec gets new errors, but not from LintDiff, but the built-in TypeSpec linter. This will be especially noticeable if there was a LintDiff violation suppressed in the "before/manual" spec, but the "after/TypeSpec-generated" spec will be flagged on the same problem  by the TypeSpec linter, because the TypeSpec linter does not read LintDiff suppressions. In such cases we will have to add TSP suppressions, the way @tjprescott explained.

Reply:
Ok the less errors case makes sense and covers my concerns.